### ISSUE TITLE
TryConversionStep executing after Using statements

### ISSUE THREAD

**Issue by mwasson74** [fluentassertions/fluentassertions#876]
### Description

It appears that the `TryConversionStep` is being executed after the `Using` statements.

### Complete minimal example reproducing the issue

```csharp
[TestMethod]
public void CompareStringToDate()
{
    //Arrange
    var classA = new ClassA { ThisIsMyDateTime = DateTime.Now };
    var classB = new ClassB { ThisIsMyDateTime = classA.ThisIsMyDateTime.ToString() };

    //Asserts
    classB.Should().BeEquivalentTo(classA,
        options => options.ExcludingMissingMembers()
            .WithAutoConversion()
            .Using<DateTime>(ctx => ctx.Subject.Should().BeCloseTo(ctx.Expectation, 1000))
            .WhenTypeIs<DateTime>());
}

private class ClassA
{
    public DateTime ThisIsMyDateTime { get; set; }
}

private class ClassB
{
    public string ThisIsMyDateTime { get; set; }
}
```

### Expected behavior:

The conversion from string to DateTime happens before the `Using` statements.

### Actual behavior:

Unit test fails with the following message:

> Expected member ThisIsMyDateTime from subject to be a System.DateTime, but found a System.String.

### Versions

* Which version of Fluent Assertions are you using?

> 5.4.1

* Which .NET runtime and version are you targeting? 

> .NET framework 4.5.2

### Additional Information
It looks like the `TryConversionStep` is placed at the end of the steps collection.

[https://github.com/fluentassertions/fluentassertions/blob/41c5e9b8a13d53794a8efe47e903c66f0c13f53e/Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyAssertionOptions.cs#L129](url)

**Comment by dennisdoomen** [comment]
I'm in doubt what would be the expected behavior. If I would provide some custom assertion rule, would one expect that to override any conversion rules. What do you think @jnyrup >

**Comment by dennisdoomen** [comment]
@jnyrup ?

**Comment by jnyrup** [comment]
I wouldn't expect custom assertion rules to override existing conversion rules.
I would expect that after failing all existing conversion rules, the code would try my custom rule.

For this particular example, here's how I imagine the code should behave:
1. If both `subject` and `expectation` are `DateTime`s, use the `Using`
2. As `WithAutoConversion` is enabled, try to convert `subject` into the type of `expectation`
2.1. If both `convertedSubject` and `expectation` are `DateTime`s, use the `Using`

**Comment by dennisdoomen** [comment]
Fixed through #1257
